swim/node.go

```diff
@@ -218,7 +218,7 @@ func (n *Node) Incarnation() int64 {
 
 func (n *Node) emit(event interface{}) {
 	for _, listener := range n.listeners {
-		go listener.HandleEvent(event)
```

Comment:
Thanks for your thoughts Dan, you haven't been the only one who has expressed worries about the safety of synchronizing the events and it has made me realize my intuition on the observer pattern has been a bit off.

At this moment, events are used for loosely coupled interactions like statting, and strongly coupled interactions updating the hashring.

I strongly believe the hashring should be synchronized with the membership. I'm all for using a different pattern for membership updates anyway.

For testing it is very useful to synchronize the events like I'm doing here. As Simmo and Nils (I believe) have suggested before, having a separate RegisterSynchronized method can help us here.

Since both sync and async events are used in the code and there is no semantic distinction at this point, I'll make a quick writeup on some possible enhancements on the event system on Monday.